Gcpe.Hub.API.Tests/ControllerTests/ActivitiesControllerTests.cs

```diff
@@ -0,0 +1,209 @@
+            result.Should().NotBeNull();
+            var models = result.Value as ICollection<Models.Activity>;
+            models.Should().NotBeNull();
+            models.Count().Should().Be(messageCount);
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(7)]
+        public void GetNewsForecast_ShouldFilterOnNumDays(int numDays)
+        {
+            var count = 3;
+            for (var i = 0; i < count; i++)
+            {
+                var testDbActivity = TestData.CreateDbActivity($"title-{i.ToString()}", "test details", i + 1);
+                context.Activity.Add(testDbActivity);
+            }
+            context.SaveChanges();
+            var controller = new ActivitiesController(context, logger.Object, mapper);
+
+            var result = controller.GetNewsForecast(numDays) as ObjectResult;
+
+            result.Should().BeOfType<OkObjectResult>();
+            var models = result.Value as ICollection<Models.Activity>;
+            models.Count().Should().Be(count);
+        }
+
+        [Fact]
+        public void GetNewsForecast_ShouldReturnBadRequest()
+        {
+            var mockContext = new Mock<HubDbContext>(this.options);
+            mockContext.Setup(m => m.Activity).Throws(new Exception());
+            var controller = new ActivitiesController(mockContext.Object, logger.Object, mapper);
+
+            var result = controller.GetNewsForecast(1) as ObjectResult;
+
+            result.StatusCode.Should().Be(400);
+            result.Should().BeOfType<BadRequestObjectResult>();
+        }
+
+        [Fact]
+        public void Post_ShouldReturnSuccess()
+        {
+            var controller = new ActivitiesController(context, logger.Object, mapper);
+            var testDbActivity = TestData.CreateDbActivity("test title", "test details", 1);
+
+            var result = controller.Post(mapper.Map<Models.Activity>(testDbActivity)) as ObjectResult;
+
+            result.StatusCode.Should().Be(201);
+            result.Should().BeOfType<CreatedAtRouteResult>();
+            var model = result.Value as Models.Activity;
+            model.Title.Should().Be("test title");
+        }
+
+        [Fact]
+        public void Get_ShouldReturnSuccess()
+        {
+            var controller = new ActivitiesController(context, logger.Object, mapper);
+            var testDbActivity = TestData.CreateDbActivity("test title", "test details", 1);
+            context.Activity.Add(testDbActivity);
+            context.SaveChanges();
+
+            var result = controller.Get(testDbActivity.Id) as ObjectResult;
+
+            result.Should().BeOfType<OkObjectResult>();
+            result.StatusCode.Should().Be(200);
+            var model = result.Value as Models.Activity;
+            model.Title.Should().Be("test title");
+        }
+
+        [Fact]
+        public void Get_ShouldReturnFail()
+        {
+            var options = new DbContextOptionsBuilder<HubDbContext>()
```

Comment:
In the previous test you didn't declare the options, you just used the same options as the other tests. Do we not need them? that would be nice and we could remove some code in my tests too